miden-tx/src/tests/kernel_tests/test_account.rs

```diff
@@ -419,11 +419,11 @@ fn test_set_map_item() {
             push.{new_value}
             push.{new_key}
             push.{item_index}
-            exec.account::set_map_item
+            call.account::set_map_item dropw
 
             # double check that on storage slot is indeed the new map
             push.{item_index}
-            exec.account::get_item
+            call.account::get_map_item 
         end
         ",
         item_index = storage_item.index,
```

Comment:
This test is the only failing one. Not sure what it is, seems to be related to the stack state after `set_map_item`

Reply:
This was fixed, but I'm not entirely sure how the fix is valid

Reply:
`set_map_item` returns `[OLD_VALUE, NEW_ROOT]` hence in this test `OLD_VALUE` is asserted for but `NEW_ROOT` is not as of now. 

We should add the following assertion:

```rust
assert_eq!(
        new_storage_map.root(),
        RpoDigest::from(process.get_stack_word(2)),
        "The returned updated value doesn't match the stored updated value",
);
```

Reply:
> `get_item()` reads the word in a given slot, regardless of the underlying type of the slot. Currently, we have two types of slots: simple values and maps. When the type of a slat is `map`, its value is actually the commitment to the map (i.e., its root).
> 
> So, if we update the map via `set_map_item` and then do `get_item` for the same storage slot, we'll get the root of the updated storage map back.
> 
> Let me know if this clarifies it.

I get the difference, but this is how I had to define `get_item` on the mock account code for it to work:

```
        export.get_item
            exec.account::get_item 
            movup.8 drop movup.8 drop movup.8 drop 
        end
```

> set_map_item returns [OLD_VALUE, NEW_ROOT] hence in this test OLD_VALUE is asserted for but NEW_ROOT is not as of now.

It seems that `::miden::account::set_map_item` returns `[OLD_MAP_ROOT, OLD_MAP_VALUE]`

Reply:
> I get the difference, but this is how I had to define `get_item` on the mock account code for it to work:
> 
> ```
>         export.get_item
>             exec.account::get_item 
>             movup.8 drop movup.8 drop movup.8 drop 
>         end
> ```


This is related to #685. Basically, we expect the above procedure to be `call`-ed and so it must leave exactly 16 items on the stack. But when we call `exec.account::get_item` it pushes 3 items onto the stack and so we need to drop them before returning from the `call`.

As discussed in #685, we should probably adapt an ABI that makes these things explicit. One general solution to this is that at the end of an account interface procedure, we should always call `std::sys::truncate_stack`. This is a little bit more expensive (in this case would take 28 cycles) - but probably not a big price to pay for a solution that would work every time.